umbraco/Umbraco-CMS
# Allow for configuration of additional languages to install and ensure the configured language is set as default.

This PR is intended to resolve [this issue](https://github.com/umbraco/Umbraco.Cloud.Issues/issues/605) with Cloud/Umbraco Deploy.

For background, a while back we introduced a feature to [allow you to configure which of the normally installed data is actually installed when Umbraco boots for the first time](https://our.umbraco.com/documentation/Reference/Configuration/InstallDefaultDataSettings/).  This prevents someone, when cloning and setting up an existing project, getting data created that has been removed by other developers working on the project.

We missed something with languages though.  Umbraco requires a default language to start up, so without `en-US` (the "default default" language) being installed, it won't start up.  

So this PR extends the `Values` option for configurable install data - used when you want to specify which of the normally set of data that is installed actually will be - to allow, for languages only, the install of _additional_ items.

And then it uses the existing configuration for the default language to set the appropriate one as a default.

So, for example, with the following configuration...

```
  "Umbraco": {
    "CMS": {
      "Global": {
        "DefaultUILanguage": "en",
      },
      "InstallDefaultData": {
        "Languages": {
          "InstallData": "Values",
          "Values": [
            "en",
            "it"
          ]
        }
      },
```

... you would get the "English (United Kingdom)" and "Italian" languages installed, with the former being the default language.  "English (United States)" would not be installed.

To Test:
 - Check the scenario described above works as expected.

Docs note: details of the ability to install additional languages should be documented on [this page](https://our.umbraco.com/documentation/Reference/Configuration/InstallDefaultDataSettings/).
